src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -242,3 +255,135 @@ def get_top_5_issues_by_count_for_file(
         ),
     )
     return raw_snql_query(request, referrer=Referrer.GITHUB_PR_COMMENT_BOT.value)["data"]
+
+
+@instrumented_task(
+    name="sentry.tasks.integrations.open_pr_comment_workflow", silo_mode=SiloMode.REGION
+)
+def open_pr_comment_workflow(pr_id: int) -> None:
```

Comment:
Can we annotate each of our checks here?

like - leave big comments calling out what we're doing in each section?